[Multiples of 3 or 5] complete solution

igorslepenkov opened the pull request.

igorslepenkov committed:
feat(multiples-3-5): add multiples of 3 or 5 problem solution

AndreyAnischik commented on src/multiple_3_5.rb:
```diff
@@ -0,0 +1,28 @@
+#   return sum
+# end
+
+# def find_sum_of_sums
```
remove comments please

AndreyAnischik commented on src/multiple_3_5.rb:
```diff
@@ -0,0 +1,28 @@
+
+def find_sum_loop
+  sum = 0
+  1000.times do |x|
```
use `reduce` instead of times

AndreyAnischik commented on src/multiple_3_5.rb:
```diff
@@ -0,0 +1,28 @@
+  1000.times do |x|
+    sum += x if (x % 3 == 0) || (x % 5 == 0)
+  end
+  return sum
```
no need to use `return`, it returns last value by default

AndreyAnischik commented on src/multiple_3_5.rb:
```diff
@@ -0,0 +1,28 @@
+  return sum
+end
+
+puts find_sum_loop()
```
add space at the end of the file

AndreyAnischik commented:
Also, I don't see tests for this piece of code

igorslepenkov committed:
refactor(multiple_3_5):  code according to comments

Add tests, remove comments, use reduse instead of times, remove return, add space to the end of file

AndreyAnischik commented on src/multiple_3_5.rb:
```diff
@@ -0,0 +1,13 @@
+module FindSumOfMultiples 
+  def self.find_sum_loop
+    (0...1000).reduce(0) do |sum, num|
+      if num % 3 == 0 || num % 5 == 0
```
use += functionality

AndreyAnischik replied:
also, you can write something like this:
```
sum += num if num % 3 == 0 || num % 5 == 0
sum
```

igorslepenkov committed:
refactor(multiple-3-5): use += and inline if statement
